server/src/main/java/org/apache/iotdb/db/sync/receiver/load/FileLoader.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.iotdb.tsfile.read.TsFileSequenceReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileLoader implements IFileLoader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileLoader.class);
+
+  private static final int WAIT_TIME = 100;
+
+  private String syncFolderPath;
+
+  private String senderName;
+
+  private BlockingQueue<LoadTask> queue = new LinkedBlockingQueue<>();
+
+  private LoadLogger loadLog;
+
+  private LoadType curType = LoadType.NONE;
+
+  private volatile boolean endSync = false;
+
+  private FileLoader(String senderName, String syncFolderPath) throws IOException {
+    this.senderName = senderName;
+    this.syncFolderPath = syncFolderPath;
+    this.loadLog = new LoadLogger(new File(syncFolderPath, SyncConstant.LOAD_LOG_NAME));
+    FileLoaderManager.getInstance().addFileLoader(senderName, this);
+    FileLoaderManager.getInstance().addLoadTaskRunner(loadTaskRunner);
+  }
+
+  public static FileLoader createFileLoader(String senderName, String syncFolderPath)
+      throws IOException {
+    return new FileLoader(senderName, syncFolderPath);
+  }
+
+  public static FileLoader createFileLoader(File syncFolder)
+      throws IOException {
+    return new FileLoader(syncFolder.getName(), syncFolder.getAbsolutePath());
+  }
+
+  private Runnable loadTaskRunner = () -> {
+    try {
+      while (true) {
+        if (queue.isEmpty() && endSync) {
+          cleanUp();
+          break;
+        }
+        LoadTask loadTask = queue.poll(WAIT_TIME, TimeUnit.MILLISECONDS);
+        if (loadTask != null) {
+          try {
+            handleLoadTask(loadTask);
+          } catch (Exception e) {
+            LOGGER.error("Can not load task {}", loadTask, e);
+          }
+        }
+      }
+    } catch (InterruptedException e) {
+      LOGGER.error("Can not handle load task", e);
+    }
+  };
+
+  @Override
+  public void addDeletedFileName(File deletedFile) {
+    queue.add(new LoadTask(deletedFile, LoadType.DELETE));
+  }
+
+  @Override
+  public void addTsfile(File tsfile) {
+    queue.add(new LoadTask(tsfile, LoadType.ADD));
+  }
+
+  @Override
+  public void endSync() {
+    if (!endSync && FileLoaderManager.getInstance().containsFileLoader(senderName)) {
+      this.endSync = true;
+    }
+  }
+
+  @Override
+  public void handleLoadTask(LoadTask task) throws IOException {
+    switch (task.type) {
+      case ADD:
+        loadNewTsfile(task.file);
+        break;
+      case DELETE:
+        loadDeletedFile(task.file);
+        break;
+      default:
+        LOGGER.error("Wrong load task type {}", task.type);
+    }
+  }
+
+  private void loadNewTsfile(File newTsFile) throws IOException {
+    if (curType != LoadType.ADD) {
+      loadLog.startLoadTsFiles();
+      curType = LoadType.ADD;
+    }
+    if (!newTsFile.exists()) {
+      LOGGER.info("Tsfile {} doesn't exist.", newTsFile.getAbsolutePath());
+      return;
```

Comment:
What? I think you mean if it will happen that the file to be loaded does not exist. Yes, actually when the system crashes during loading. In this case, tsfiles are loaded but it's not logged to the log file, which will result in the recovery when restarting the system.